mlrun/api/api/endpoints/projects.py

```diff
@@ -1,79 +1,67 @@
-from operator import attrgetter
 from http import HTTPStatus
 
-from fastapi import APIRouter, Depends, Response
+from fastapi import APIRouter, Depends, Response, Header
 from sqlalchemy.orm import Session
 
 from mlrun.api import schemas
 from mlrun.api.api import deps
-from mlrun.api.api.utils import log_and_raise
-from mlrun.api.db.sqldb.helpers import to_dict as db2dict
-from mlrun.api.utils.singletons.db import get_db
+from mlrun.api.utils.singletons.projects_manager import get_projects_manager
 
 router = APIRouter()
 
 
 # curl -d '{"name": "p1", "description": "desc", "users": ["u1", "u2"]}' http://localhost:8080/project
-@router.post("/project")
-def add_project(
-    project: schemas.ProjectCreate, db_session: Session = Depends(deps.get_db_session)
+@router.post("/projects", response_model=schemas.Project)
+def create_project(
+    project: schemas.Project, db_session: Session = Depends(deps.get_db_session)
 ):
-    project_id = get_db().add_project(db_session, project.dict())
-    return {
-        "id": project_id,
-        "name": project.name,
-    }
+    get_projects_manager().create_project(db_session, project)
+    return get_projects_manager().get_project(db_session, project.name)
 
 
 # curl -d '{"name": "p1", "description": "desc", "users": ["u1", "u2"]}' -X UPDATE http://localhost:8080/project
-@router.post("/project/{name}")
-def update_project(
-    project: schemas.ProjectUpdate,
+@router.put("/projects/{name}", response_model=schemas.Project)
+def store_project(
```

Comment:
Moving the name discussion here:
Options: store, update, put, upsert, create_or_update
I don't think we're married to store, we do have it in several "production" places already, but we can easily keep supporting these while moving towards the place we want - here specifically it's new so definitely we can do what we want
Already had this discussion with @theSaarco here https://github.com/mlrun/mlrun/pull/532#discussion_r523722571
I was in favor of "upsert", I still prefer it over "put", and "update" is not an option IMO (I prefer "store" over it), but I do understand the claim it's too DB-oriented 
I must say I'm convinced we should simply go with "create_or_update" - simple and clear

Reply:
@theSaarco + @yaronha you're welcome to give your thoughts

Reply:
I feel like I don't have the whole story here (why "put" is an option but "update" not? is this more closely correlated to CRUD? or REST verbs?) - taking it offline

Reply:
It all boils down to the actual operation being done - per client requirements (@yaronha), these APIs perform "create or update". That's why "update" is not enough - it implies create is not supported. Given the logic performed, **store** works, **put** is a bit shaky IMO since people have their understanding of what put does which may or may not align with our actual implementation. **Upsert** seems (at least to me) to be too close to the actual DB operation involved, which is probably not an issue for the SQLDB function, but once it is used  elsewhere it becomes very much out of context.

Reply:
@Hedingber  - Shouldn't this turn to `create_or_update` ?